### ISSUE TITLE
SSD wear information is not correct in csm node resources query

### ISSUE THREAD

**Issue by tgooding** [IBM/CAST#325]
**Describe the bug**
Wear statistics aren't being accurately reported.  

**To Reproduce**
Run csm_node_resources_query_all on a cluster with SSDs that have some wear.

**Expected behavior**
SSD wear reported in CSM should closely match the reported wear from the SSD firmware.

**Screenshots**
```
bash-4.2$ /opt/ibm/csm/bin/csm_node_resources_query_all | grep wear
      wear_lifespan_used: 0.000000
      wear_lifespan_used: 0.000000
      wear_lifespan_used: 0.000000
```

Then executing on one of the compute nodes:
```
bash-4.2$ sudo nvme smart-log /dev/nvme0n1 | grep percentage_used
percentage_used                     : 2%
```

**Environment (please complete the following information):**
bbdev cluster

**Comment by mew2057** [comment]
@besawn

**Comment by besawn** [comment]
@NickyDaB 

This is a known issue. CSM inventory collection does not collect this metric or several other SSD metrics because inventory collection runs infrequently, so these values will not be guaranteed to be up to date if they are gathered by inventory collection.

When SSD inventory collection was implemented, the need for a new mechanism to collect this SSD metric data was discussed, but we don't have a final resolution. We need to invent a new mechanism to populate these SSD metric values and make sure they get updated periodically.

**Comment by tgooding** [comment]
This doesn't feel like its isolated to a periodic update issue.  On my cluster, the database is wiped at each installation and so i'd think inventory collection would run immediately when csmd is started.  e.g., I see ssd_serial_number populated but wear data isn't.

**Comment by besawn** [comment]
Exactly, it is never collected because it is not "inventory" it is a "metric" and we need to invent something to collect it.

**Comment by mew2057** [comment]
What's the status on this one?

**Comment by tgooding** [comment]
GPLv3 license in that code.  @NickyDaB : if you want to go the ioctl() route, we can send you details - but you'd be somewhat baking in kernel level structures into your userland code.  They normally don't change, but its not a guarantee.

**Comment by besawn** [comment]
This is fixed now via PR #570. Please re-open if you see any additional problems after testing with the latest code.
